src/operator/nn/cudnn/cudnn_convolution-inl.h

```diff
@@ -512,18 +511,15 @@ class CuDNNConvolutionOp {
     }
   }
 
-  void SelectAlgo(const Context& ctx,
+  void SelectAlgo(const RunContext& rctx,
                   const std::vector<TShape>& in_shape,
                   const std::vector<TShape>& out_shape,
                   cudnnDataType_t cudnn_forward_compute_type,
                   cudnnDataType_t cudnn_backward_compute_type) {
     if (!CuDNNConvAlgoReg::Get()->Find(param_, in_shape, out_shape, dtype_,
                                        cudnn_forward_compute_type, cudnn_backward_compute_type,
-                                       SMArch(ctx.dev_id), &forward_algo_, &back_algo_,
+                                       SMArch(rctx.ctx.dev_id), &forward_algo_, &back_algo_,
                                        &back_algo_w_)) {
-      // Not in algo registry, must determine via *Get*() or *Find*()
-      Engine::VarHandle var = Engine::Get()->NewVariable();
-      Engine::Get()->PushAsync([=](RunContext rctx, Engine::CallbackOnComplete on_complete) {
         mshadow::Stream<gpu> *s = rctx.get_stream<gpu>();
```

Comment:
nit: indentation?

Reply:
The indentation of this piece of code is indeed not right. I changed it in a previous commit but it resulted in huge code difference that makes code review much harder. I can make the indentation right either after this PR is merged or after everyone finishes reviewing the current code changes. :)